### PR TITLE
fix: dockerfile build failed

### PR DESCRIPTION
**Pull request by zll600**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.

- use registry.yarnpkg.com instead of `registry.taobao.com`
- update Dockerfile use copy of local file instead of use `git clone`
- update ci file to remove `APISIX_DASHBOARD_VERSION=master`

**Related issues**

fix/resolve https://github.com/apache/apisix-dashboard/issues/2917

**Checklist:**

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by zll600** [comment]
@monkeyDluffy6017 @nic-chen @bzp2010 Could you help review when you have time?

File: Dockerfile [hunk 1]
```diff
@@ -16,11 +16,10 @@
 #
 FROM alpine:latest as pre-build
 
-ARG APISIX_DASHBOARD_VERSION=master
+COPY . /usr/local/apisix-dashboard
```
Comment (starsz) [inline Dockerfile:19]: The "./docs/en/latest/deploy-with-docker.md"  is using this environment, don't forget to modify this.

Comment (zll600) [reply Dockerfile:19]: OK, I deleted them

**Review by starsz — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

**Comment by zll600** [comment]
@monkeyDluffy6017 Should we backport this to `release/3.0`?

**Comment by zll600** [comment]
The [`bashboard CI` in `apisix-build-tools`](https://github.com/api7/apisix-build-tools/blob/f7fd733ed4479a985df2bdb58de961ce60257b9c/.github/workflows/package-apisix-dashboard-rpm-el7.yml#L22)  will fail if we don't fix it on `release/3.0` 
https://github.com/api7/apisix-build-tools/actions/runs/7865663864/job/21458910075

**Pull request merged**